back-end-projects/OntSynHandler/src/main/java/com/github/ontio/thread/TxHandlerThread.java

```diff
@@ -685,35 +708,25 @@ private void handleOep8TransferTx(JSONArray stateArray, int txType, String txHas
             TxDetail txDetail = generateTransaction("", "", "", ConstantParam.ZERO, txType, txHash, blockHeight,
                     blockTime, indexInBlock, confirmFlag, "", gasConsumed, indexInTx, EventTypeEnum.Others.type(), contractAddress, payer, calledContractHash);
 
-            ConstantParam.BATCHBLOCKDTO.getTxDetails().add(txDetail);
-            ConstantParam.BATCHBLOCKDTO.getTxDetailDailys().add(TxDetail.toTxDetailDaily(txDetail));
-            ConstantParam.BATCHBLOCKDTO.getOep8TxDetails().add(TxDetail.toOep8TxDetail(txDetail));
+            updateTxDetailsOep8(txDetail);
             return;
         }
 
         String action = new String(Helper.hexToBytes((String) stateArray.get(0)));
-        String fromAddress = (String) stateArray.get(1);
-        String toAddress = (String) stateArray.get(2);
+        String fromAddress = parseAddress((String) stateArray.get(1));
+        String toAddress = parseAddress((String) stateArray.get(2));
         String tokenId = (String) stateArray.get(3);
         JSONObject oep8Obj = (JSONObject) ConstantParam.OEP8MAP.get(contractAddress + "-" + tokenId);
-        if ("00".equals(fromAddress) && stateSize == 2) {
-            // mint方法即增加发行量方法, 区分标志：fromAddress为“00”，同时stateSize为2
+
+        if ("00".equals(fromAddress) || "00".equals(toAddress)) {
             Oep8 oep8 = Oep8.builder()
                     .contractHash(contractAddress)
-                    .tokenId((String) stateArray.get(3))
-                    .totalSupply(commonService.getOep8TotalSupply(tokenId))
+                    .tokenId(tokenId)
+                    .totalSupply(commonService.getOep8TotalSupply(contractAddress, tokenId))
                     .build();
-            //在子线程直接更新，不批量更新
             oep8Mapper.updateByPrimaryKeySelective(oep8);
         }
 
-        if (40 == fromAddress.length()) {
-            fromAddress = Address.parse(fromAddress).toBase58();
-        }
-        if (40 == toAddress.length()) {
-            toAddress = Address.parse(toAddress).toBase58();
-        }
-
         BigDecimal eventAmount = new BigDecimal(Helper.BigIntFromNeoBytes(Helper.hexToBytes((String) stateArray.get(4))).longValue());
```

Comment:
Please fix inherited narrowing bug:
```suggestion
         BigDecimal eventAmount = new BigDecimal(Helper.BigIntFromNeoBytes(Helper.hexToBytes((String) stateArray.get(4))));
```
https://docs.oracle.com/javase/7/docs/api/java/math/BigDecimal.html#BigDecimal%28java.math.BigInteger%29

Reply:
Done.